File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.py
class Solution:
    def addToArrayForm(self, num: List[int], k: int) -> List[int]:
        
        for i in range(len(num) - 1, -1, -1):
            k, d = divmod(k, 10)
            carry, num[i] = divmod(num[i] + d, 10)
            k += carry
            if not k:
                break
                
        if k:
            num = list(map(int, str(k))) + num
        return num

File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.cpp
class Solution {
public:
    vector<int> addToArrayForm(vector<int>& num, int k) {
        int n = num.size(), curr = k;
        vector<int> ans;
        int i = n;
        
        while (--i >= 0 || curr > 0){
            if (i >= 0) curr += num[i];
            ans.push_back(curr % 10);
            curr /= 10;
        }
        
        reverse(ans.begin(), ans.end());
        return ans;
    }
};